marshmallow_recipe/serialization.py

```diff
@@ -151,3 +151,15 @@ def dump_many_v2(data: list[Any], *, naming_case: NamingCase | None = None) -> l
     dump_many = dump_many_v2
 
 EmptySchema = m.Schema
+
+
+def _extract_type(data: Any, cls: type | None) -> type:
+    if hasattr(data, "__orig_class__"):
```

Comment:
just in case, why we may not have this attribute in place

https://github.com/python/cpython/issues/117744
https://github.com/python/cpython/issues/101688

Reply:
please check PR description, section **Extracting generic instance type**